### PR TITLE
Supports overriding the default paths required for custom installs

### PR DESCRIPTION
**Pull request by LePresidente**
Implements: https://github.com/crowdsecurity/cs-openresty-bouncer/issues/16

I have tested on my setup and works

This requires this as well for the Nginx Proxy Manager Integration https://github.com/NginxProxyManager/docker-nginx-full/pull/7
https://github.com/crowdsecurity/lua-cs-bouncer/pull/27

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -35,27 +58,51 @@ requirement() {
...
+        sed "s/=.*//g" ./config/config_example.conf  > /tmp/config_example.conf.raw
+        if grep -vf /tmp/crowdsec.conf.raw /tmp/config_example.conf.raw ; then
+            grep -vf /tmp/crowdsec.conf.raw /tmp/config_example.conf.raw > /tmp/config_example.newvals
+            cp "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf" "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf.bak"
+            #Make sure we start on a new line.
+             echo "" >>"${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+            grep -f /tmp/config_example.newvals /tmp/crowdsec/config/config_example.conf >> "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+        fi
+    fi
+    # Not sure why couldn't patch the path using envsubst
```
Comment (he2ss) [inline install.sh:88]: Maybe there is no envsubst in the NPM docker image ? 
Don't forget to remove not important comments please.

Comment (LePresidente) [reply install.sh]: yea this make sense i'll update the docker image where this is required.

File: install.sh [hunk 2]
```diff
@@ -9,12 +9,35 @@ PKG="apt"
 PACKAGE_LIST="dpkg -l"
 SSL_CERTS_PATH="/etc/ssl/certs/ca-certificates.crt"
 
+#Accept cmdline arguments to overwrite options.
+while [[ $# -gt 0 ]]
+do
+    case $1 in
+	    --NGINX_CONF_DIR=*)
+		    NGINX_CONF_DIR="${1#*=}"
```
[the remark this answers is not in the document]
Comment (LePresidente) [reply install.sh]: fixed in latest commit

File: install.sh [hunk 3]
```diff
@@ -35,27 +58,51 @@ requirement() {
...
+    #Don't overwrite the existing file
+    if [ ! -f "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf" ]; then
+        if [ -z ${DOCKER} ]; then
+            SUFFIX=$(tr -dc A-Za-z0-9 </dev/urandom | head -c 8)
+            API_KEY=$(cscli bouncers add "crowdsec-openresty-bouncer-${SUFFIX}" -o raw) 
+            API_KEY=${API_KEY} CROWDSEC_LAPI_URL="http://127.0.0.1:8080" envsubst < ./config/config_example.conf > "${CONFIG_PATH}/crowdsec-openresty-bouncer.conf"
+            echo "New API key generated to be used in '${CONFIG_PATH}/crowdsec-openresty-bouncer.conf'"
+        else
+            #Docker doesn't support envsubst by default
+            API_KEY="1234567890abcdef"
```
[the remark this answers is not in the document]
Comment (LePresidente) [reply install.sh]: changed this a little, the script should check if cscli is preset before it attempts to auto generate a key. If not it should just leave the key blank.

**Review by he2ss — APPROVED** [review]

**Pull request merged**